app/src/main/java/org/p2p/wallet/history/repository/remote/TransactionDetailsRpcRepository.kt

```diff
@@ -1,40 +1,82 @@
-    private val transactionDetailsNetworkMapper: TransactionDetailsNetworkMapper,
-    private val userLocalRepository: UserLocalRepository
+    private val userInteractor: UserInteractor,
 ) : TransactionDetailsRemoteRepository {
 
-    override suspend fun getTransactions(signatures: List<String>): List<TransactionDetails> {
-        val requestsBatch = signatures.map { signature ->
-            val encoding = buildMap {
-                this[RpcConstants.REQUEST_PARAMETER_KEY_ENCODING] =
-                    RpcConstants.REQUEST_PARAMETER_VALUE_JSON_PARSED
-            }
-            val params = listOf(signature, encoding)
-
+    override suspend fun getTransactions(
+        userPublicKey: String,
+        transactionSignature: List<RpcTransactionSignature>
+    ): List<TransactionDetails> {
+        val encoding = buildMap {
+            this[RpcConstants.REQUEST_PARAMETER_KEY_ENCODING] =
+                RpcConstants.REQUEST_PARAMETER_VALUE_JSON_PARSED
+            this[RpcConstants.REQUEST_PARAMETER_KEY_COMMITMENT] =
+                RpcConstants.REQUEST_PARAMETER_VALUE_CONFIRMED
+        }
+        val requestsBatch = transactionSignature.map { signature ->
+            val params = listOf(signature.signature, encoding)
             RpcRequest(method = RpcConstants.REQUEST_METHOD_VALUE_GET_CONFIRMED_TRANSACTIONS, params = params)
         }
 
-        return rpcApi.getConfirmedTransactions(requestsBatch)
-            .map { it.result }
-            .let {
-                withContext(dispatchers.io) {
-                    transactionDetailsNetworkMapper.fromNetworkToDomain(it) { mintAddress ->
-                        userLocalRepository.findTokenData(mintAddress)?.mintAddress.orEmpty()
+        val transactions = rpcApi.getConfirmedTransactions(requestsBatch).map { it.result }
+
+        return fromNetworkToDomain(userPublicKey, transactions).onEach { transactionDetails ->
+            transactionDetails.status =
+                transactionSignature.first { it.signature == transactionDetails.signature }.status
+        }
+    }
+
+    private fun fromNetworkToDomain(
+        tokenPublicKey: String,
+        transactions: List<ConfirmedTransactionRootResponse>
+    ): List<TransactionDetails> {
+        val transactionDetails = mutableListOf<TransactionDetails>()
+        transactions.forEach { transaction ->
+            try {
+                val signature = transaction.transaction?.getTransactionId() ?: return@forEach
+                when {
+                    OrcaSwapInstructionParser.isTransactionContainsOrcaSwap(transaction) -> {
+                        val orcaSwapDetails =
+                            OrcaSwapInstructionParser.parse(signature = signature, transactionRoot = transaction)
+                        transactionDetails.add(orcaSwapDetails.getOrThrow())
+                    }
+                    SerumSwapInstructionParser.isTransactionContainsSerumSwap(transaction) -> {
```

Comment:
please remove static calls here and inject parser directly

Reply:
For what ?, it's just objects, i don't see reasons convert them to classes

Reply:
It would be easier to mock their behavior later in unit tests.
instead of duck taping, we could just mock any of parsers to return true/false for their `is*` methods to check every type